Source build of 1.9.0 on macOS ARM: make testall failures in testset ambiguous

Built committ `5e08c20c70`. This failure is new since `1fb28ad876` which built and passed `make testall`.

Test was run with an isolated `JULIA_DEPOT_PATH`

<details>
<summary>click for versioninfo details</summary>

```
Julia Version 1.9.0-DEV.528
Commit 5e08c20c70 (2022-05-10 23:28 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.4.0)
  uname: Darwin 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64 arm
  CPU: Apple M1 Max:
                 speed         user         nice          sys         idle          irq
       #1-10    24 MHz     191717 s          0 s     116929 s    4116858 s          0 s
  Memory: 64.0 GB (43704.296875 MB free)
  Uptime: 44254.0 sec
  Load Avg:  1.19140625  1.68310546875  2.81640625
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
Environment:
  JULIA_DEPOT_PATH = /Users/foo/Desktop
  XPC_FLAGS = 0x0
  PATH = /Users/foo/.pyenv/bin:/opt/homebrew/opt/grep/libexec/gnubin:/Users/foo/bin:/Users/foo/.local/bin:/opt/homebrew/bin:/Users/foo/Library/Python/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/MacPorts/EmacsMac.app/Contents/MacOS:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Apple/usr/bin:/Users/foo/perl5/perlbrew/bin:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/bin:/Users/foo/.local/bin:/opt/homebrew/opt/fzf/bin:/Users/foo/.emacs.d/bin
  TERM = xterm-256color
  HOME = /Users/foo
  PERLBREW_HOME = /Users/foo/.perlbrew
  PERLBREW_MANPATH = /Users/foo/perl5/perlbrew/perls/perl-5.35.5/man
  PERLBREW_PATH = /Users/foo/perl5/perlbrew/bin:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/bin
  MANPATH = /opt/local/share/man:/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/Library/TeX/Distributions/.DefaultTeX/Contents/Man:/opt/X11/share/man:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/man:/Library/Apple/usr/share/man:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/share/man:/Applications/Xcode.app/Contents/Developer/usr/share/man:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man:
  HOMEBREW_PREFIX = /opt/homebrew
  HOMEBREW_CELLAR = /opt/homebrew/Cellar
  HOMEBREW_REPOSITORY = /opt/homebrew
  INFOPATH = /opt/homebrew/share/info:
```

</details>

<details>
<summary>click for testall summary</summary>

```
% gmake testall 
                                                                                                                                                                                                                                                                                                                                                      21:50
Whitespace check found no issues.
cp /Users/foo/applications/juliad/usr/lib/julia/sys.dylib /Users/foo/applications/juliad/local.dylib
/Users/foo/applications/juliad/usr/bin/julia -J /Users/foo/applications/juliad/local.dylib -e 'true'
rm /Users/foo/applications/juliad/local.dylib
    JULIA test/all
Running parallel tests with:
  nworkers() = 8
  nthreads() = 1
  Sys.CPU_THREADS = 8
  Sys.total_memory() = 64.000 GiB
  Sys.free_memory() = 37.355 GiB

Test Summary:                             |     Pass  Fail  Broken     Total      Time
  Overall                                 | 40547809     4  352662  40900475  15m56.8s
    LinearAlgebra/bunchkaufman            |     5688                    5688     33.1s
    LinearAlgebra/eigen                   |      512                     512     40.3s
    LinearAlgebra/lapack                  |      802                     802     17.8s
    LinearAlgebra/qr                      |     4705                    4705     58.5s
    LinearAlgebra/svd                     |      566                     566     32.8s
    LinearAlgebra/schur                   |      496                     496   1m23.8s
    LinearAlgebra/tridiag                 |     1541                    1541     25.8s
    LinearAlgebra/special                 |     2942                    2942   1m49.8s
    LinearAlgebra/lu                      |     1368                    1368     55.5s
    LinearAlgebra/cholesky                |     2509                    2509   1m02.1s
    LinearAlgebra/dense                   |     8475                    8475   2m33.0s
    LinearAlgebra/generic                 |      622                     622     20.7s
    LinearAlgebra/bidiag                  |     4736                    4736   1m45.7s
    LinearAlgebra/lq                      |     2938                    2938     23.4s
    LinearAlgebra/uniformscaling          |      446                     446     35.5s
    LinearAlgebra/matmul                  |     1487                    1487   3m05.2s
    LinearAlgebra/pinv                    |      500                     500      5.7s
    LinearAlgebra/hessenberg              |      631                     631     27.5s
    LinearAlgebra/givens                  |     1847                    1847      3.8s
    LinearAlgebra/adjtrans                |      347                     347     13.1s
    LinearAlgebra/ldlt                    |        8                       8      0.7s
    LinearAlgebra/blas                    |     1600                    1600     26.3s
    LinearAlgebra/factorization           |       80            16        96      2.1s
    compiler/validation                   |       28                      28      0.3s
    ambiguous                             |      104     4       2       110      3.1s
    compiler/ssair                        |       40                      40      1.8s
    compiler/irpasses                     |      139             4       143      1.5s
    compiler/inline                       |      177             1       178      2.0s
    compiler/contextual                   |       12                      12      1.4s
    compiler/inference                    |     1207             2      1209      8.6s
    compiler/AbstractInterpreter          |       12                      12      2.8s
    compiler/codegen                      |      172                     172      8.3s
    compiler/EscapeAnalysis/interprocedural |       32             4        36      3.5s
    compiler/EscapeAnalysis/local         |      347            21       368      6.9s
    strings/search                        |      876                     876      3.1s
    strings/basic                         |    87689                   87689      5.3s
    LinearAlgebra/diagonal                |     2873                    2873   2m25.3s
    LinearAlgebra/structuredbroadcast     |      670                     670     24.6s
    unicode/utf8                          |       19                      19      0.1s
    strings/io                            |    12764                   12764      1.7s
    worlds                                |       88                      88      1.2s
    strings/types                         |  2302691                 2302691      2.0s
    keywordargs                           |      151                     151      0.9s
    strings/util                          |     1147                    1147      5.2s
    atomics                               |     3448                    3448      7.4s
    subtype                               |   337678            19    337697      7.4s
    triplequote                           |       29                      29      0.0s
    char                                  |     1639                    1639      1.4s
    intrinsics                            |      301                     301      1.4s
    hashing                               |    12519                   12519      7.5s
    iobuffer                              |      209                     209      0.7s
    staged                                |       65                      65      1.2s
    dict                                  |   144424                  144424     12.9s
    numbers                               |  1578845             3   1578848     33.0s
    tuple                                 |      657                     657      4.3s
    offsetarray                           |      488             3       491     20.0s
    core                                  |  8445937             3   8445940     45.6s
    reduce                                |     8588                    8588     10.2s
    intfuncs                              |   228060                  228060     14.6s
    simdloop                              |      240                     240      0.9s
    vecelement                            |      678                     678      3.5s
    reducedim                             |     1089             6      1095     34.0s
    subarray                              |   318316                  318316   1m25.5s
    LinearAlgebra/symmetric               |     2823                    2823   2m60.0s
    copy                                  |      533                     533      1.3s
    arrayops                              |     2033             2      2035     55.3s
    rational                              |    98639             1     98640     11.6s
    functional                            |       98                      98      2.8s
    fastmath                              |      946                     946      4.4s
    LinearAlgebra/triangular              |    37908                   37908   4m54.6s
    path                                  |      373            12       385      0.7s
    ordering                              |       37                      37      1.3s
    operators                             |    13046                   13046      3.5s
    parse                                 |    16098                   16098      2.5s
    gmp                                   |     2444                    2444      3.4s
    ccall                                 |   165125             1    165126     12.4s
    math                                  |  1867032                 1867032     20.4s
    loading                               |   152939                  152939     14.3s
    backtrace                             |       39             1        40      1.3s
    exceptions                            |       70                      70      1.2s
    file                                  |     1074                    1074      5.8s
    version                               |     2453                    2453      0.5s
    namedtuple                            |      216                     216      1.6s
    spawn                                 |      252             4       256     14.3s
    abstractarray                         |    55990         24795     80785   1m07.5s
    mpfr                                  |     1137             1      1138      8.3s
    floatapprox                           |       49                      49      0.5s
    read                                  |     3872                    3872     19.7s
    reflection                            |      425                     425      3.4s
    regex                                 |      130                     130      1.3s
    complex                               |     8479             2      8481      9.3s
    sysinfo                               |        4                       4      0.1s
    env                                   |       95                      95      0.3s
    float16                               |   762093                  762093      3.6s
    combinatorics                         |      170                     170      3.6s
    mod2pi                                |       80                      80      0.2s
    rounding                              |   150012                  150012      2.8s
    euler                                 |       12                      12      0.7s
    client                                |        5                       5      0.9s
    errorshow                             |      237                     237      3.3s
    iterators                             |    86611                   86611     52.0s
    goto                                  |       19                      19      0.1s
    llvmcall                              |       19                      19      0.3s
    llvmcall2                             |       20                      20      0.1s
    ryu                                   |    31215                   31215      1.0s
    some                                  |       72                      72      0.6s
    meta                                  |       69                      69      1.0s
    stacktraces                           |       48                      48      1.2s
    docs                                  |      238                     238      2.2s
    sets                                  |     3625             1      3626     13.6s
    show                                  |   128886             8    128894     19.7s
    binaryplatforms                       |      341                     341      2.9s
    enums                                 |       99                      99      1.9s
    atexit                                |       40                      40      4.8s
    ranges                                | 12110690        327682  12438372     27.0s
    interpreter                           |        3                       3      1.1s
    checked                               |     1239                    1239      4.6s
    int                                   |   736188                  736188      6.5s
    bitset                                |      195                     195      1.5s
    broadcast                             |      514                     514     50.3s
    error                                 |       33                      33      1.1s
    boundscheck                           |                             None      4.7s
    osutils                               |       43                      43      0.1s
    sorting                               |    24052             9     24061   1m13.1s
    iostream                              |       50                      50      0.7s
    secretbuffer                          |       27                      27      0.9s
    specificity                           |      175                     175      0.1s
    cartesian                             |      343             3       346      5.2s
    misc                                  |  1282213                 1282213     29.9s
    corelogging                           |      235                     235      4.0s
    syntax                                |     1637             1      1638      6.2s
    reinterpretarray                      |      418                     418     13.3s
    smallarrayshrink                      |       36                      36      0.3s
    opaque_closure                        |       66             1        67      0.5s
    filesystem                            |        6                       6      0.2s
    channels                              |      256                     256     19.4s
    asyncmap                              |      307                     307      9.3s
    missing                               |      574             1       575     13.6s
    bitarray                              |   911542                  911542   1m58.0s
    SparseArrays/basetests                |       28                      28      2.1s
    LibGit2/online                        |       10                      10      2.7s
    download                              |                             None     31.1s
    Dates/accessors                       |  7723858                 7723858      3.7s
    Dates/adjusters                       |     3149                    3149      2.4s
    Dates/query                           |     1004                    1004      0.5s
    Dates/periods                         |      953                     953      8.9s
    Dates/ranges                          |   350637                  350637      9.8s
    Dates/rounding                        |      315                     315      1.3s
    Dates/types                           |      232                     232      0.8s
    Dates/io                              |      332                     332      7.1s
    Dates/arithmetic                      |      385                     385      3.7s
    Dates/conversions                     |      161                     161      0.7s
    SparseArrays/higherorderfns           |     7145             1      7146   1m10.4s
    ArgTools                              |      180                     180      4.8s
    Base64                                |     2031                    2031      1.5s
    Artifacts                             |     1452                    1452      3.3s
    CompilerSupportLibraries_jll          |        4                       4      0.0s
    CRC32c                                |      664                     664      0.4s
    DelimitedFiles                        |       89                      89      3.6s
    LibGit2/libgit2                       |      754                     754   1m03.3s
    Future                                |                             None      0.0s
    GMP_jll                               |        1                       1      0.4s
    InteractiveUtils                      |      281                     281     10.4s
    LLVMLibUnwind_jll                     |        7                       7      0.2s
    LazyArtifacts                         |        4                       4      1.8s
    LibCURL                               |        6                       6      1.2s
    LibCURL_jll                           |        1                       1      0.0s
    LibGit2_jll                           |        2                       2      0.7s
    LibSSH2_jll                           |                             None      0.0s
    LibUV_jll                             |        1                       1      0.3s
    LibUnwind_jll                         |                             None      0.3s
    Libdl                                 |       79             1        80      0.5s
    Logging                               |       40                      40      0.7s
    MPFR_jll                              |        1                       1      0.2s
    Markdown                              |      257                     257      2.5s
    MbedTLS_jll                           |        1                       1      0.0s
    cmdlineargs                           |      279             5       284   2m23.4s
    MozillaCACerts_jll                    |        1                       1      0.0s
    NetworkOptions                        |     3518                    3518      0.6s
    OpenBLAS_jll                          |        1                       1      0.0s
    OpenLibm_jll                          |        1                       1      0.2s
    PCRE2_jll                             |        2                       2      0.2s
    Mmap                                  |      140                     140     15.8s
    SparseArrays/sparsevector             |    10348             4     10352   1m56.7s
    Printf                                |     1014                    1014      7.4s
    REPL                                  |     1424             3      1427     17.4s
    Downloads                             |      238                     238   1m07.3s
    floatfuncs                            |      232                     232   2m49.3s
    Serialization                         |      129             1       130      4.1s
    Sockets                               |      172                     172      4.8s
    Random                                |   204835                  204835     28.1s
    SHA                                   |      107                     107     41.0s
    SuiteSparse_jll                       |        1                       1      0.8s
    FileWatching                          |      634                     634   1m48.3s
    Statistics                            |      804                     804     34.2s
    TOML                                  |      415             8       423      5.5s
    UUIDs                                 |     1029                    1029      0.2s
    Profile                               |      113                     113   1m08.3s
    Zlib_jll                              |        1                       1      0.3s
    dSFMT_jll                             |        1                       1      0.3s
    libLLVM_jll                           |        1                       1      0.2s
    libblastrampoline_jll                 |        1                       1      0.0s
    nghttp2_jll                           |        1                       1      0.0s
    p7zip_jll                             |        1                       1      0.0s
    Unicode                               |      795                     795      2.6s
    Test                                  |      454            17       471     16.3s
    Tar                                   |     3161            11      3172     26.1s
    SparseArrays/sparse                   |     4354                    4354   3m39.9s
    LinearAlgebra/addmul                  |     6948                    6948   7m19.8s
    SuiteSparse                           |     1293                    1293   1m08.2s
    Pkg                                   |     2449             2      2451   7m27.4s
    precompile                            |      155                     155     18.0s
    SharedArrays                          |      114                     114     11.8s
    threads                               |       40                      40   1m23.4s
    Distributed                           |       15                      15   1m37.7s
    stress                                |      118                     118      7.5s
    FAILURE

The global RNG seed was 0xd11e0741a07eee9126497629bd7e37de.

Error in testset ambiguous:
Test Failed at /Users/foo/applications/juliad/test/ambiguous.jl:177
  Expression: sig1 === sig2
   Evaluated: Tuple{Type{Main.Test66Main_LinearAlgebra_matmul.TestMatmul.BrokenInt}, Any} === Tuple{Type{T}, AbstractChar} where T<:Union{AbstractChar, Number}
Error in testset ambiguous:
Test Failed at /Users/foo/applications/juliad/test/ambiguous.jl:177
  Expression: sig1 === sig2
   Evaluated: Tuple{Type{Main.Test66Main_LinearAlgebra_matmul.TestMatmul.BrokenInt}, Any} === Tuple{Type{T}, Base.TwicePrecision} where T<:Number
Error in testset ambiguous:
Test Failed at /Users/foo/applications/juliad/test/ambiguous.jl:177
  Expression: sig1 === sig2
   Evaluated: Tuple{Type{Main.Test66Main_LinearAlgebra_matmul.TestMatmul.BrokenInt}, Any} === Tuple{Type{T}, T} where T<:Number
Error in testset ambiguous:
Test Failed at /Users/foo/applications/juliad/test/ambiguous.jl:180
  Expression: good
ERROR: LoadError: Test run finished with errors
in expression starting at /Users/foo/applications/juliad/test/runtests.jl:93
gmake[1]: *** [Makefile:25: all] Error 1
gmake: *** [Makefile:554: testall] Error 2
```

</details>

--- a/stdlib/LinearAlgebra/test/matmul.jl
+++ b/stdlib/LinearAlgebra/test/matmul.jl
@@ -825,17 +825,6 @@ end
     @test Matrix{Int}(undef, 2, 0) * Matrix{Int}(undef, 0, 3) == zeros(Int, 2, 3)
 end
 
-struct BrokenInt <: Number
-   i::Int
-end
-Base.:*(::BrokenInt, ::BrokenInt) = BrokenInt(42)
-Base.:+(::BrokenInt, ::BrokenInt) = BrokenInt(42)
-Base.zero(::BrokenInt) = BrokenInt(0)
-Base.conj(b::BrokenInt) = b.i == 42 ? b : error()
-@testset "matmul uninit memory #40481" begin
-    @test fill(BrokenInt(42), 10,100)' * fill(BrokenInt(42), 100,10)' == fill(BrokenInt(42), 100, 100)
-end
-
 @testset "3-arg *, order by type" begin
     x = [1, 2im]
     y = [im, 20, 30 + 40im]

